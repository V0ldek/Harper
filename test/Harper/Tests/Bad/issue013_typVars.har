x :: a -> b;

f :: a ;
f = {
  (g :: c -> d)= x ;
  (h :: e -> f)= x ;
  match g h {
    (i :: a) => return i ;
    }
  } ;

main :: Bool;
main = f;