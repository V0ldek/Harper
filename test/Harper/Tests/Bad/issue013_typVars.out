Error: expression `h` has invalid type 'e& -> f&'; expected 'c&'.
During evaluation of:
 g h

Located at line 7 column 9


Execution terminated with an error: type error.